addons/api/addon/abilities/role.js

```diff
@@ -26,4 +35,22 @@ export default class RoleAbility extends ModelAbility {
   get canRemovePrincipals() {
     return this.hasAuthorizedAction('remove-principals');
   }
+
+  /**
+   * Returns the ability check for a principal model instance.
+   * Valid role principal models are: User, Group, and ManagedGroup.
+   * Throws an error if the principal type is invalid.
+   * @type {boolean}
+   * @throws {InvalidRolePrincipalTypeError}
+   */
+  get canReadPrincipal() {
+    const type = this.model.constructor.modelName;
+    const typeIsValid =
+      type === 'user' || type === 'group' || type === 'managed-group';
```

Comment:
```suggestion
      type === 'user' || type === 'group' || type === 'managed group';
```

Reply:
The input for this is an actual Ember Data model instance, in which the model name is `managed-group`.